src/Microsoft.Diagnostics.Monitoring.RestServer/DiagnosticServices.cs

```diff
@@ -114,82 +120,37 @@ private static DumpType MapDumpType(Models.DumpType dumpType)
             if (processKey.HasValue)
             {
-                return GetSingleProcessInfo(
-                    processInfos,
-                    processKey);
+                filterOptions = DiagProcessFilter.FromProcessKey(processKey.Value);
             }
-
-            // Short-circuit for when running in a Docker container.
-            if (RuntimeInfo.IsInDockerContainer)
+            else
             {
-                try
-                {
-                    IProcessInfo processInfo = GetSingleProcessInfo(
-                        processInfos,
-                        DockerEntrypointProcessFilter);
-
-                    using var timeoutSource = new CancellationTokenSource(DockerEntrypointWaitTimeout);
-
-                    var client = new DiagnosticsClient(processInfo.EndpointInfo.Endpoint);
-                    await client.WaitForConnectionAsync(timeoutSource.Token);
-
-                    return processInfo;
-                }
-                catch
-                {
-                    // Process ID 1 doesn't exist, didn't advertise in connect mode, or is not a .NET process.
-                }
+                filterOptions = DiagProcessFilter.FromConfiguration(_defaultProcessOptions.CurrentValue);
             }
 
-            return GetSingleProcessInfo(
-                processInfos,
-                processKey: null);
+            return GetProcessAsync(filterOptions, token);
         }
 
-        private IProcessInfo GetSingleProcessInfo(IEnumerable<IProcessInfo> processInfos, ProcessKey? processKey)
+        private async Task<IProcessInfo> GetProcessAsync(DiagProcessFilter processFilterConfig, CancellationToken token)
         {
-            if (processKey.HasValue)
+            //Short circuit when we are missing default process config
+            if (!processFilterConfig.Filters.Any())
             {
-                if (processKey.Value.RuntimeInstanceCookie.HasValue)
-                {
-                    Guid cookie = processKey.Value.RuntimeInstanceCookie.Value;
-                    processInfos = processInfos.Where(info => info.EndpointInfo.RuntimeInstanceCookie == cookie);
-                }
-
-                if (processKey.Value.ProcessId.HasValue)
-                {
-                    int pid = processKey.Value.ProcessId.Value;
-                    processInfos = processInfos.Where(info => info.EndpointInfo.ProcessId == pid);
-                }
-
-                string processName = processKey.Value.ProcessName;
-                if (!string.IsNullOrEmpty(processName))
-                {
-                    processInfos = processInfos.Where(info => string.Equals(info.ProcessName, processName, StringComparison.OrdinalIgnoreCase));
-                }
+                throw new InvalidOperationException("No default process configuration has been set.");
             }
+            IEnumerable<IProcessInfo> matchingProcesses = await GetProcessesAsync(processFilterConfig, token);
```

Comment:
I would suggest a ToArray or ToList on this so that you avoid multiple evaluations of the underlying LINQ expression. Or have GetProcessesAsync does this.